test/test_queue_monkeypatch.py

```diff
@@ -0,0 +1,32 @@
+from __future__ import absolute_import
+
+import unittest
+
+import mock
+
+import urllib3
+from urllib3.exceptions import EmptyPoolError
+from urllib3.packages.six.moves import queue
+
+class BadError(Exception):
+    """
+    This should not be raised.
+    """
+    pass
+
+
+class TestMonkeypatchResistance(unittest.TestCase):
+    """
+    Test that connection pool works even with a monkey patched Queue module,
+    see obspy/obspy#1599, kennethreitz/requests#3742, shazow/urllib3#1061.
+    """
+    def test_queue_monkeypatching(self):
+        with mock.patch.object(queue, 'Empty', BadError):
+            http = urllib3.HTTPConnectionPool(host="localhost", block=True)
+            first_conn = http._get_conn(timeout=1)
+            with self.assertRaises(EmptyPoolError):
```

Comment:
Given that we don't use `second_conn` why can't we just do:

```py
self.assertRaises(EmptyPoolError, http._get_conn, timeout=1)
```

This will avoid the unittest2 shenanigans altogether